types/layui-src/index.d.ts

```diff
@@ -1,745 +1,18 @@
-        predefine?: boolean | undefined;
-        colors?: string[] | undefined;
-        size?: 'lg' | 'sm' | 'xs' | undefined;
-        change?: ((color: any) => void) | undefined;
-        done?: ((color: any) => void) | undefined;
-    }
-
-    interface ColorPicker {
-        render(option: ColorPickerOption): ColorPicker;
-    }
-
-    interface SliderOption {
-        elem?: string | HTMLElement | undefined;
-        type?: 'default' | 'vertical' | undefined;
-        mix?: number | undefined;
-        max?: number | undefined;
-        range?: boolean | undefined;
-        value?: number | number[] | undefined;
-        step?: number | undefined;
-        showstep?: boolean | undefined;
-        tips?: boolean | undefined;
-        input?: boolean | undefined;
-        height?: number | undefined;
-        disabled?: boolean | undefined;
-        theme?: string | undefined;
-        setTips?: ((value: number | number[]) => string) | undefined;
-        change?: ((value: number | number[]) => string) | undefined;
-    }
-
-    interface Slider {
-        render(option: SliderOption): Slider;
-        setValue(value1: number, value2?: number): void;
-    }
-
-    interface Layedit {
-        build(id: string, options?: EditOption): any;
-
-        set(options: EditOption): void;
-
-        getContent(index: number): string;
-
-        getText(index: number): string;
-
-        sync(index: number): void;
-
-        getSelection(index: number): string;
-    }
-
-    interface EditOption {
-        tool?: string[] | undefined;
-        hideTool?: string[] | undefined;
-        height?: number | string | undefined;
-        uploadImage?: { url: string; type: string } | undefined;
-    }
-
-    interface Laytpl {
-        (tpl: string): TplObject;
-    }
-
-    interface TplObject {
-        render(data: object, callback: (str: string) => void): any;
-    }
-
-    let layer: Layer;
-    let element: Element;
-    let form: Form;
-    let laydate: Laydate;
-    let layedit: Layedit;
-    let laypage: Laypage;
-    let laytpl: Laytpl;
-    let carousel: Carousel;
-    let table: Table;
-    let upload: Upload;
-    let rate: Rate;
-    let flow: Flow;
-    let util: Util;
-    let colorpicker: ColorPicker;
-    let slider: Slider;
-    let $: JQueryStatic;
-    function code(option?: CodeOption): void;
-    function tree(option: TreeOption): void;
-    function use(mods: string | string[], callback: (...args: any) => any): any;
-}
+// Type definitions for layui-src 2.6
+// Project: https://github.com/sentsin/layui/releases/tag/v2.6.8
+// Definitions by: javabitar <https://github.com/javabitar>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 3.1
+
+/// <reference path="misc.d.ts" />
+/// <reference path="Layui.d.ts" />
+
+declare global {
+    interface Window {
+        layui: Layui;
+        lay: Layui.Lay;
+        layer: Layui.Layer;
+    }
+}
+
+export = layui;

```

Comment:
```suggestion
```
Why this change to support CommnJS (or UMD-like) modules?
I could be wrong, but I think `layyi-src` is just a global JS script, expected to be bundled in any supported way and there is no ESM or CommonJS usage/definitions in native package.
It can still be used in CJS that way I believe, without this export:
```ts
/// <reference types="layui-src" />
import "jquery";

$.when($.ready).then(() => {
  layui.use(["layer", "form"], () => {
    const layer = layui.layer,
      form = layui.form;
    layer.msg("Hello World");
  });
});
```